File: VB/ViewModels/UserViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports DevExpress.DevAV.Chat.Model

Namespace DXHtmlMessengerSample.ViewModels

    Public Class UserViewModel
        Inherits UserInfoViewModel

        Protected Friend Overrides ReadOnly Property ServiceKey() As String
            Get
                Return "UserInfoDialog"
            End Get
        End Property
        Public Shared Function Create(ByVal user As UserInfo) As UserViewModel
            Return ViewModelSource.Create(Function() New UserViewModel(user))
        End Function
        '
        Protected Sub New(ByVal user As UserInfo)
            MyBase.New(user)
        End Sub
        Public Sub LogOff()
            Dim messenger = Me.GetParentViewModel(Of MessengerViewModel)()
            If messenger IsNot Nothing Then
                messenger.LogOff()
            End If
        End Sub
        Public Sub Close()
            CloseDocument()
        End Sub
    End Class
End Namespace

File: CS/ViewModels/UserViewModel.cs
namespace DXHtmlMessengerSample.ViewModels {
    using DevExpress.DevAV.Chat.Model;
    using DevExpress.Mvvm.POCO;

    public class UserViewModel : UserInfoViewModel {
        protected internal override string ServiceKey {
            get { return "UserInfoDialog"; }
        }
        public static UserViewModel Create(UserInfo user) {
            return ViewModelSource.Create(() => new UserViewModel(user));
        }
        //
        protected UserViewModel(UserInfo user)
            : base(user) {
        }
        public void LogOff() {
            CloseDocument();
            var messenger = this.GetParentViewModel<MessengerViewModel>();
            if(messenger != null) messenger.LogOff();
        }
        public void Close() {
            CloseDocument();
        }
    }
}